### PR TITLE
usable image viewer plugin

### PR DESCRIPTION
**Pull request by jmonster**
most of these changes are from format-on-save triggering; I added the following to make it work:

```js
var decodeAttributeByName = (fromClass, name) => {
  const encodedJSON = fromClass.getAttribute(name);
  const decodedJSON = encodedJSON
    ?.replace(/&quot;/g, '"')
    ?.replace(/&#39;/g, "'");

  try {
    return JSON.stringify(JSON.parse(decodedJSON));
  } catch (e) {
    return JSON.stringify({});
  }
};

var triggerEvent = (fromClass, data) => {
  const event = new CustomEvent("custom-change", {
    detail: data,
    bubbles: true,
    composed: true,
  });

  fromClass.dispatchEvent(event);
};
```

and commented out
```js
// JOHNNY commented this out because it serves no purpose but to interrupt installation (at the moment?)
// window.customElements.define('outerbase-plugin-configuration-$PLUGIN_ID', OuterbasePluginConfiguration_$PLUGIN_ID)
```

image

### CODE REVIEW

**Review by caleb-mabry — CHANGES_REQUESTED** [review]
It's important if you're going to put this in the global scope, to append _$PLUGIN_ID to the end of it. If you don't, you'll have naming collisions with other plugins that might also use this variable name

**Comment by jmonster** [comment]
@caleb-mabry can you be more explicit about what part you want changed? the function names? ELI5

**Comment by caleb-mabry** [comment]
@jmonster Yeah, sorry.
`decodeAttributeByName` and `triggerEvent` needs `_$PLUGIN_ID` appended to the end of the name. Since neither of those are within any of the classes, once this file is loaded on the page, if another plugin has the same variable name defined, it'll cause issues. You could move both of those functions inside of each class and call them with `this.` to get around that, but it's a little more duplicated code. Your call on what you want.
